src/pkg/selectors/groups.go

```diff
@@ -309,7 +310,76 @@ func (s *groups) LibraryItems(libraries, items []string, opts ...option) []Group
 // -------------------
 // ItemInfo Factories
 
-// TODO
+// MessageCreator produces one or more groups channelMessage info scopes.
+// Matches any channel message created by the specified user.
+// If any slice contains selectors.Any, that slice is reduced to [selectors.Any]
+// If any slice contains selectors.None, that slice is reduced to [selectors.None]
+// If any slice is empty, it defaults to [selectors.None]
+func (s *GroupsRestore) MessageCreator(creator string) []GroupsScope {
+	return []GroupsScope{
+		makeInfoScope[GroupsScope](
+			GroupsChannelMessage,
+			GroupsInfoChannelMessageCreator,
+			[]string{creator},
+			filters.In),
+	}
+}
+
+// MessageCreatedAfter produces a channel message created-after info scope.
+// Matches any message where the creation time is after the timestring.
+// If the input equals selectors.Any, the scope will match all times.
+// If the input is empty or selectors.None, the scope will always fail comparisons.
+func (s *GroupsRestore) MessageCreatedAfter(timeStrings string) []GroupsScope {
```

Comment:
I haven't added `CreatedAfter` and other funcs for Libraries, but was wondering if we need this filter separately for different services.

Reply:
You're welcome to try and consolidate the interanls if you want, but I don't think there's much left to reduce/reuse.

The primary reason for duplicating the funcs themeselves (ie: CreatedAfter) is that we want the selector to provide a curated list of scopes that are only applicable to that specific resource.  And, of course, that we need to provide selector-type-specific consts to the underlying scope.  Providing a generic "CreatedAfter" scope wouldn't allow for either of those.

Reply:
Hmm, my concern was mostly around the UX. When restoring messages(which I know is not possible as of now), should we have `MessageCreatedAfter` override `CreatedAfter` as we will have to restore the attachments referenced in the message?

Reply:
Ah, I grok.  No, let's not do that at this time.  Per prior sync with Georgi, we're not going to restore attachments at the same time as messages.  It's fine to keep them separated.